src/mower_logic/src/mower_logic/behaviors/Behavior.h

```diff
@@ -158,6 +158,7 @@ class Behavior {
 
     virtual uint8_t get_sub_state() = 0;
     virtual uint8_t get_state() = 0;
+    virtual int16_t get_current_area() = 0;
```

Comment:
Since it's specific to mowing behaviour why not just add it in that header?

Reply:
so leave it in MowingBehavior.h, but strip it out of Behavior.h and all the non-mowing behavior headers and cpp's?

Reply:
and probably need to dynamic_cast currentBehavior to MowingBehavior when you want to call the MowingBehavior method